persistence/in-memory/src/main/java/org/eclipse/tractusx/wallet/stub/storage/memory/MemoryStorage.java

```diff
@@ -119,4 +136,16 @@ public Optional<KeyPair> getKeyPair(String bpn) {
     public Optional<DidDocument> getDidDocument(String bpn) {
         return Optional.ofNullable(DID_DOCUMENT_STORE.get(bpn));
     }
+
+    @Override
+    public List<CustomCredential> getVcIdAndTypesByHolderBpn(String holderBpn) {
+        return HOLDER_CREDENTIAL_STORE.values().stream().filter(credential -> {
+            Map<String, String> credentialSubject = (Map<String, String>) credential.get(Constants.CREDENTIAL_SUBJECT_CAMEL_CASE);
```

Comment:
Should we add a defensive check here to avoid a possible `ClassCastException` exception?  Correct me if i am wrong here...

Possible modification in below
```suggestion
            Object subject = credential.get(Constants.CREDENTIAL_SUBJECT_CAMEL_CASE);
if (!(subject instanceof Map<?, ?> credentialSubject)) return false;
```

Reply:
Addressed with https://github.com/Cofinity-X/entry-ssi-dim-wallet-stub/pull/3/commits/4d16061a851a054bfa21880a910aea4cc78e3062